gk/fib.c

```diff
@@ -60,45 +65,113 @@ gk_lpm_add_ipv6_route(uint8_t *ip,
  * callers must ensure that the entry is not being used.
  */
 static inline int
-initialize_ether_cache(struct ether_cache *eth_cache)
+clear_ether_cache(struct ether_cache *eth_cache)
 {
-	memset(eth_cache, 0, sizeof(*eth_cache));
+	memset(eth_cache->fields_to_clear, 0, sizeof(*eth_cache) -
+		offsetof(struct ether_cache, fields_to_clear));
+
+	if (rte_atomic32_read(&eth_cache->ref_cnt) != 0) {
```

Comment:
This test must be `!= 1`, otherwise it's going to miss buggy cases in which `== 0` and wrongly log correct cases in which `== 1`.